src/glue/jobs/s3_to_json.py

```diff
@@ -84,12 +86,30 @@ def transform_object_to_array_of_objects(
         try:
             key_value = key_type(k)
         except ValueError:
-            logger.warning(value_error, k,key_type)
+            logger.warning(
+                    value_error, k, key_type,
+                    extra={
+                        **logger_context,
+                        "event.kind": "alert",
+                        "event.category": ["configuration"],
+                        "event.type": ["change"],
```

Comment:
Thoughts on the `deletion` event.type?
https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html#ecs-event-type-deletion
vs
https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html#ecs-event-type-change

I'm leaning towards `deletion` myself because we are destroying the `key_value` in this case rather than a transformation into some other useful value.

Reply:
I see what you mean. Deletion does make more sense here.

Reply:
@BryanFauble I'm having trouble differentiating how we would log two very different types of events. For example:

1. Failing to delete a configuration setting or parameter:
```
logger.error(
    "Failed to delete configuration or parameter",
    extra={
        "event.kind": "alert",
        "event.category": ["configuration"],
        "event.type": ["deletion"],
        "event.outcome": "failure"
    }
)                      
```

2. Failing to do some operation on data, resulting in that data being dropped/deleted:
```
logger.error(
    "Failed to do an operation on data. We will drop/delete this data instead.",
    extra={
        "event.kind": "alert",
        "event.category": ["configuration"],
        "event.type": ["deletion"],
        "event.outcome": "??????"
    }
)
# Code to drop data goes here           
```
I'm starting to think that `"event.type": ["change"]` and `"event.outcome": "failure"` makes more sense here.

Reply:
I am thinking of these log messages as "What is the result of the thing I tried to do".
For your example what do you think of:
```
try:
    # Do something
except:
    logger.error(
        "Failed to do an operation on data. We will drop/delete this data instead.",
        extra={
            "event.kind": "alert",
            "event.category": ["configuration"],
            "event.type": ["deletion"],
            "event.outcome": "failure"
        }
    )
    try:
        # Do the delete
    except:
        logger.error(
            "Failed to delete configuration or parameter",
            extra={
                "event.kind": "alert",
                "event.category": ["configuration"],
                "event.type": ["error"],
                "event.outcome": "failure"
            }
        )    
```

Reply:
Taking a hint from the `event.outcome` ECS docs:
> Note that when a single transaction is described in multiple events, each event may populate different values of event.outcome, according to their perspective.

I think this makes more sense:
```
try:
    # Do something
except:
    logger.error(
        "Failed to do an operation on data",
        extra={
            "event.kind": "alert",
            "event.category": ["configuration"],
            "event.type": ["change"],
            "event.outcome": "failure"
        }
    )
    try:
        # Do the delete
        logger.warning(
            "Deleted some data",
            extra={
                "event.kind": "alert",
                "event.category": ["configuration"],
                "event.type": ["deletion"],
                "event.outcome": "success"
            }
        )
    except:
        logger.error(
            "Failed to delete data",
            extra={
                "event.kind": "alert",
                "event.category": ["configuration"],
                "event.type": ["deletion"],
                "event.outcome": "failure"
            }
        )  
```

Reply:
I like it